lib/IDE/Utils.cpp

```diff
@@ -1268,20 +1311,18 @@ bool swift::ide::isDynamicCall(Expr *Base, ValueDecl *D) {
   if (Base->isSuperExpr())
     return false;
 
-  // `SomeType.staticOrClassMethod()`
+  // `SomeType.staticOrClassMethod()` spelled directly, so this must be a ref
+  // to this exact decl.
   if (isa<TypeExpr>(Base))
     return false;
 
-  // `type(of: foo).staticOrClassMethod()`, not "dynamic" if the instance type
-  // is a struct/enum or if it is a class and the function is a static method
-  // (rather than a class method).
+  // `type(of: foo).staticOrClassMethod()`. A static method may be "dynamic"
+  // here, but not if the instance type is a struct/enum.
```

Comment:
What about actors and final classes? I see that you removed the case below. Doesn’t the same apply to them?

Reply:
They are already covered by the early return